crates/shared/src/transport/http.rs

```diff
@@ -147,6 +147,10 @@ impl BatchTransport for HttpTransport {
 
         async move {
             let _guard = metrics.on_request_start("batch");
+            let _guards: Vec<_> = calls
+                .iter()
+                .map(|call| metrics.on_request_start(method_name(call)))
+                .collect();
```

Comment:
This mixes up some things in the metrics.
1. We record the whole batch and the individual requests making it up but there is no way to distinguish this. If I make one batch request with one internal request we count 2 total requests even though 1 is more accurate.
2. It is not useful to record timing information for the requests making up a batch. They all take the same duration that the batched request takes. If a request takes one second and I make 3 in a row then I get an average of 1 second. But if I make a batch request that includes these three then we get a 3 second average.

I feel it would be more useful to duplicate

```rust
    /// Number of completed RPC requests for ethereum node.
    #[metric(labels("method"))]
    requests_complete: prometheus::IntCounterVec,
```
into a new metrics whose only responsibility is to count individual requests inside of batch requests. `inner_batch_requests_complete` .

Reply:
I agree that we should separate timing information for batch requests as it will be low signal (each request in the batch will take the total time as you suggested). This could lead to us thinking "why is `eth_blockNumber` taking so long?" when in fact it only does because it is included in a batch with 1000 complicated `eth_call`s).

Reply:
To be clear, I don't think we need timing for the inner requests of the batches at all. We only need to count them. So I would only duplicate the one metric I mentioned and not the histogram or in-flight.

Reply:
How about not duplicating counters and instead just add an argument to `on_request_start()` which allows to discard timing data? That seems like a low effort solution on the rust side as well as the grafana side.

Reply:
On_request_start does a bunch of things, most of which are not useful for the inner batch calls. Feels cleaner to me to handle the inner batch requests completely separately because all we want to do there is increment one prometheus counter. No need for DropGuard logic either because that is only used for timing the request.

Reply:
>  I don't think we need timing for the inner requests of the batches at all

I think we should keep timing information for the batch, but not for the inner calls. This is what I meant with my ambiguous "should separate timing information for batch requests". I think we are arguing for the same thing.

Reply:
I would:
- Keep track of success/failure for all calls (regular and inner batch) per method
- Keep track of timming for regular calls and for the complete batch, but not for the inner batch calls themselves.